c10/core/TensorImpl.cpp

```diff
@@ -990,6 +990,12 @@ void TensorImpl::set_sizes_and_strides(
   refresh_contiguous();
 }
 
+void TensorImpl::set_storage_offset(c10::SymInt storage_offset) {
+  if (!extra_meta_) {
+    extra_meta_ = std::make_unique<ExtraMeta>();
+  }
+  extra_meta_->storage_offset_ = storage_offset.clone();
+}
```

Comment:
No, this is not sound. set_sizes_and_strides has a bunch of logic for routing to the normal non `extra_meta_` case when there are no symbolics anywhere, and it needs to be replicated here

Reply:
or better yet, we continue to use `set_sizes_and_strides` so we only need one implementation of these functions